src/modules/Elsa.Workflows.Runtime/Services/DefaultWorkflowDefinitionStorePopulator.cs

```diff
@@ -114,47 +117,8 @@ private async Task AddOrUpdateCoreAsync(MaterializedWorkflow materializedWorkflo
         if (existingDefinitionVersion != null)
             workflowDefinitionsToSave.Add(existingDefinitionVersion);
 
-        // If the workflow being added is configured to be the latest version, then we need to reset the current latest version.
-        if (workflow.Publication.IsLatest)
-        {
-            // Reset current latest definitions.
-            var filter = new WorkflowDefinitionFilter
-            {
-                DefinitionId = definitionId,
-                VersionOptions = VersionOptions.Latest
-            };
-            var latestWorkflowDefinitions = (await _workflowDefinitionStore.FindManyAsync(filter, cancellationToken)).ToList();
-
-            // If the latest definitions contains definitions with the same ID then we need to replace them with the latest workflow definitions.
-            SyncExistingCopies(latestWorkflowDefinitions, workflowDefinitionsToSave);
-
-            foreach (var latestWorkflowDefinition in latestWorkflowDefinitions)
-            {
-                latestWorkflowDefinition.IsLatest = false;
-                workflowDefinitionsToSave.Add(latestWorkflowDefinition);
-            }
-        }
-
-        // If the workflow being added is configured to be the published version, then we need to reset the current published version.
-        if (workflow.Publication.IsPublished)
-        {
-            // Reset current published definitions.
-            var filter = new WorkflowDefinitionFilter
-            {
-                DefinitionId = definitionId,
-                VersionOptions = VersionOptions.Published
-            };
-            var publishedWorkflowDefinitions = (await _workflowDefinitionStore.FindManyAsync(filter, cancellationToken)).ToList();
-
-            // If the published workflow definitions contains definitions with the same ID as definitions in the latest workflow definitions, then we need to replace them with the latest workflow definitions.
-            SyncExistingCopies(publishedWorkflowDefinitions, workflowDefinitionsToSave);
-
-            foreach (var publishedWorkflowDefinition in publishedWorkflowDefinitions)
-            {
-                publishedWorkflowDefinition.IsPublished = false;
-                workflowDefinitionsToSave.Add(publishedWorkflowDefinition);
-            }
-        }
+        await UpdateIsLatest();
+        await UpdateIsPublished();
```

Comment:
As an optimization, do you think it would be possible to do these two calls after all workflows have been imported? Perhaps, while a workflow is being imported, you should capture information like "this workflow is published" in local variables so that you can be specific about what workflow definition versions must be published.

Reply:
As discussed offline with @raymonddenhaan , we will revisit this later.

Reply:
Created enhancement #5075 for this.